controllers/fenceagentsremediation_controller.go

```diff
@@ -226,14 +233,19 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		}
 
 		r.Log.Info("Build fence agent command line", "Fence Agent", far.Spec.Agent, "Node Name", node.Name)
-		faParams, err := buildFenceAgentParams(far)
+		secretParams, err := r.collectSecretParams(far, ctx)
 		if err != nil {
-			r.Log.Error(err, "Invalid shared or node parameters from CR", "Node Name", node.Name, "CR Name", req.Name)
+			r.Log.Error(err, "Failed collecting secrets data", "Node Name", node.Name, "CR Name", req.Name)
+			return emptyResult, err
+		}
+		faParams, err := buildFenceAgentParams(far, secretParams)
+		if err != nil {
+			r.Log.Error(err, "Invalid credential/shared/node parameter from CR", "Node Name", node.Name, "CR Name", req.Name)
```

Comment:
NIT: For better logging, we can replace `credential` by `secret` as there are no more `credentialParam` in this PR. It would be nicer if it followed the order of parameter priority as mentioned at https://github.com/medik8s/fence-agents-remediation/pull/171/files#r2089036850

Reply:
Ah Sorry, I  got it now.
Initially I thought you were referring to the order [here](https://github.com/medik8s/fence-agents-remediation/pull/171/files#diff-05147ee3f0d225b5a60754b8b9bff2c569deb0a9d7eaddae33af680291d493a1R248)